aspnetcore/tutorials/signalr-blazor-webassembly.md

```diff
@@ -0,0 +1,357 @@
+
+# [Visual Studio for Mac](#tab/visual-studio-mac)
+
+1. In a command shell, execute the following command:
+
+   ```dotnetcli
+   dotnet new blazorwasm --hosted --output BlazorSignalRApp
+   ```
+
+1. In Visual Studio for Mac, open the project by navigating to the project folder and opening the project's solution file (*.sln*).
+
+# [.NET Core CLI](#tab/netcore-cli/)
+
+In a command shell, execute the following command:
+
+```dotnetcli
+dotnet new blazorwasm --hosted --output BlazorSignalRApp
+```
+
+---
+
+## Add the SignalR client library
+
+# [Visual Studio](#tab/visual-studio/)
+
+1. In **Solution Explorer**, right-click the **BlazorSignalRApp.Server** project and select **Manage NuGet Packages**.
+
+1. In the **Manage NuGet Packages** dialog, confirm that the **Package source** is set to *nuget.org*.
+
+1. Enable the **Include prerelease** option.
+
+1. With **Browse** selected, type "Microsoft.AspNetCore.SignalR.Client" in the search box.
+
+1. In the search results, select the `Microsoft.AspNetCore.SignalR.Client` package and select **Install**.
+
+1. If the **Preview Changes** dialog appears, select **OK**.
+
+1. If the **License Acceptance** dialog appears, select **I Accept** if you agree with the license terms.
+
+1. **Repeat these steps** for the **BlazorSignalRApp.Client** project.
+
+# [Visual Studio Code](#tab/visual-studio-code/)
+
+In the **Integrated Terminal** (**View** > **Terminal** from the toolbar), execute the following commands:
+
+```console
+dotnet add Server package Microsoft.AspNetCore.SignalR.Client
+dotnet add Client package Microsoft.AspNetCore.SignalR.Client
+```
+
+# [Visual Studio for Mac](#tab/visual-studio-mac)
+
+1. In the **Solution** sidebar, right-click the **BlazorSignalRApp.Server** project and select **Manage NuGet Packages**.
+
+1. In the **Manage NuGet Packages** dialog, confirm that the source drop-down is set to *nuget.org*.
+
+1. Enable the **Show pre-release packages** option at the bottom of the dialog.
+
+1. With **Browse** selected, type "Microsoft.AspNetCore.SignalR.Client" in the search box.
+
+1. In the search results, select the check box next to the `Microsoft.AspNetCore.SignalR.Client` package and select **Add Package**.
+
+1. If the **License Acceptance** dialog appears, select **Accept** if you agree with the license terms.
+
+1. **Repeat these steps** for the **BlazorSignalRApp.Client** project.
+
+# [.NET Core CLI](#tab/netcore-cli/)
+
+In a command shell, execute the following commands:
+
+```dotnetcli
+cd BlazorSignalRApp
+dotnet add Server package Microsoft.AspNetCore.SignalR.Client
+dotnet add Client package Microsoft.AspNetCore.SignalR.Client
+```
+
+---
+
+## Add a SignalR hub
+
+In the **BlazorSignalRApp.Server** project, create a *Hubs* (plural) folder and add the following `ChatHub` class (*Hubs/ChatHub.cs*):
+
+```csharp
+using System;
```

Comment:
Should we clean this sample up a little if we're going to put it in a doc?

Like displaying when the connection connects and having a disconnect button etc.

Also, shouldn't code usually go in .cs files and be linked in?

Reply:
I defer to DR on the code, since he supplied it.

WRT a snippet ... Let's hold on that. I don't have feedback on whether or not we should have a downloadable sample. I'll put this into a snippet if I hear back that we don't need a formal sample.